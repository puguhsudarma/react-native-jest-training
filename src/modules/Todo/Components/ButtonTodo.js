// @flow

import React from 'react';
import {
    View,
    Text,
    TouchableHighlight,
} from 'react-native';
import { styles as s } from 'react-native-style-tachyons';

type ButtonTodoPropTypes = {
    action: () => void,
    disabled: boolean,
};

const ButtonTodo = (props: ButtonTodoPropTypes) => {
    return (
        <View>
            <TouchableHighlight onPress={props.action} disabled={props.disabled}>
                <View style={[s.aic, s.jcc, {
                    padding: 10,
                    backgroundColor: 'blue',
                }]}>
                    <Text style={{
                        textAlign: 'center',
                        fontWeight: '600',
                    }}>
                        {'Simpan'}
                    </Text>
                </View>
            </TouchableHighlight>
        </View>
    );
};

export default ButtonTodo;
