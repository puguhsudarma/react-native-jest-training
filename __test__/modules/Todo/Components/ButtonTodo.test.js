import React from "react";
import renderer from 'react-test-renderer';
import ShallowRenderer from 'react-test-renderer/shallow';
import ButtonTodo from "../../../../src/modules/Todo/Components/ButtonTodo";

describe("Test Button Todo Render", () => {
  // mock function
  const actionFn = jest.fn();

  it("Render deep button todo correctly", () => {
    const tree = renderer.create(<ButtonTodo action={actionFn} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("Render shallow button todo correctly", () => {
    const shallow = new ShallowRenderer();
    shallow.render(<ButtonTodo action={actionFn} />);
    const tree = shallow.getRenderOutput();
    expect(tree).toMatchSnapshot();
  });
});
