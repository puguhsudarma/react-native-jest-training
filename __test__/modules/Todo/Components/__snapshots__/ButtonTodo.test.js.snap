// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Button Todo Render Render deep button todo correctly 1`] = `
<View>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "borderRadius": 0,
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Array [
          undefined,
          undefined,
          Object {
            "backgroundColor": "blue",
            "padding": 10,
          },
        ]
      }
    >
      <Text
        style={
          Object {
            "fontWeight": "600",
            "textAlign": "center",
          }
        }
      >
        Simpan
      </Text>
    </View>
  </View>
</View>
`;

exports[`Test Button Todo Render Render shallow button todo correctly 1`] = `
<Component>
  <UniButton
    action={[MockFunction]}
    activeOpacity={0.5}
    borderRadius={0}
    borderless={false}
    disabled={false}
    foregroundRipple={false}
    highlightColor="rgba(255,255,255,0.15)"
    noDebounce={false}
    useTouchableHightlight={false}
  >
    <Component
      style={
        Array [
          undefined,
          undefined,
          Object {
            "backgroundColor": "blue",
            "padding": 10,
          },
        ]
      }
    >
      <Component
        style={
          Object {
            "fontWeight": "600",
            "textAlign": "center",
          }
        }
      >
        Simpan
      </Component>
    </Component>
  </UniButton>
</Component>
`;
